packages/grafana-ui/src/components/Link/Link.tsx

```diff
@@ -1,21 +1,116 @@
+import { css } from '@emotion/css';
 import React, { AnchorHTMLAttributes, forwardRef } from 'react';
 import { Link as RouterLink } from 'react-router-dom';
 
-import { locationUtil, textUtil } from '@grafana/data';
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
 
-export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {}
+import { useTheme2 } from '../../themes';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+
+export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** If the link will redirect users to a page in or out Grafana */
+  external?: boolean;
+  /** Specify the icon name to be used when the link is not an inline element. If this is undefined but the link is external and not inline, the link icon will be shown. */
```

Comment:
This one is quite confusing to follow now, but lets see what the decision is about showing icons in inline or not to see if we can simplify this 👍